velox/functions/prestosql/aggregates/SumDataSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,215 @@
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void
+  updateOneAccumulator(char* const group, vector_size_t row, int64_t rowSize) {
+    if (decoded_.isNullAt(row)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set sum(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = checkedPlus(current, rowSize);
+  }
+
+  void
+  doUpdate(char** groups, const SelectivityVector& rows, const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+    estimateSerializedSizes(arg, rows, rows.countSelected());
+    vector_size_t sizeIndex = 0;
+    rows.applyToSelected([&](auto row) {
+      updateOneAccumulator(groups[row], row, elementSizes_[sizeIndex++]);
+    });
+  }
+
+  // Estimate the sizes of first numToProcess selected elements in vector.
+  void estimateSerializedSizes(
+      VectorPtr vector,
+      const SelectivityVector& rows,
+      vector_size_t numToProcess) {
+    elementSizes_.resize(numToProcess);
+    std::fill(elementSizes_.begin(), elementSizes_.end(), 0);
+    elementIndices_.resize(numToProcess);
+    elementSizePtrs_.resize(numToProcess);
+
+    vector_size_t i = 0;
+    rows.testSelected([&](auto row) {
+      elementIndices_[i] = IndexRange{row, 1};
+      elementSizePtrs_[i] = &elementSizes_[i];
+      return ++i < numToProcess;
+    });
+
+    getVectorSerde()->estimateSerializedSize(
+        vector,
+        folly::Range(elementIndices_.data(), elementIndices_.size()),
+        elementSizePtrs_.data());
+  }
+
+  void doUpdateSingleGroup(
+      char* group,
+      const SelectivityVector& rows,
+      const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+    estimateSerializedSizes(arg, rows, rows.countSelected());
+    vector_size_t sizeIndex = 0;
+    rows.applyToSelected([&](auto row) {
+      updateOneAccumulator(group, row, elementSizes_[sizeIndex++]);
+    });
+  }
+};
+
+exec::AggregateRegistrationResult registerMaxSizeForStats(
+    const std::string& name) {
+  std::vector<std::shared_ptr<exec::AggregateFunctionSignature>> signatures;
+
+  signatures.push_back(exec::AggregateFunctionSignatureBuilder()
+                           .typeVariable("T")
+                           .returnType("BIGINT")
+                           .intermediateType("BIGINT")
+                           .argumentType("T")
+                           .build());
+
+  return exec::registerAggregateFunction(
+      name,
+      std::move(signatures),
+      [name](
+          core::AggregationNode::Step step,
+          const std::vector<TypePtr>& argTypes,
+          const TypePtr& resultType) -> std::unique_ptr<exec::Aggregate> {
+        VELOX_CHECK_EQ(argTypes.size(), 1, "{} takes only one argument", name);
+        auto inputType = argTypes[0];
+
+        return std::make_unique<SumDataSizeForStatsAggregate>(resultType);
+      });
+}
+
+} // namespace
+
+void registerSumDataSizeForStatsAggregate(const std::string& prefix) {
```

Comment:
Can we rename registerMaxSizeForStats to registerMaxDataSizeForStats to be consistent? Also shall we add internal prefix?

Reply:
Renamed 

As for internal prefix, from the reference PR https://github.com/facebookincubator/velox/pull/1797, they already put out a note for this function for public usage, so we can keep it as it is and have a followup discussion around if we want to make this function private.

Reply:
Yeah, can you help file a task or issue to track this. It seems to me that these stats function are for internal use. Thanks!

Reply:
These are for internal use in Presto, but... Velox is a separate library and cannot make these functions internal.

Reply:
To keep them internal, they need to be moved to Prestissimo codebase.

Reply:
Is it possible to add some code like https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/types/PrestoToVeloxExpr.cpp#L36 (though in a new mapAggregateName function) to do the mapping ?